open-craft/taxonomy-connector
# feat: xblocks skill api

**Description:**
Adds api to fetch xblocks related to a skill as well as fetch skills related to an XBlock.

**Test instructiions**

1. Clone this repo in `<devstack_base_dir>/src/ and checkout this branch.
2. Verify tests using `make requirements validate` in repo folder.
3. Start lms and discovery service using `make lms-up discovery-up`.
4. Open discovery shell: `make discovery-shell`.
5. Install this repo using `pip install -e /src/taxonomy-connector`.
6. Run `./manage.py migrate`.
7. Run `./manage.py shell`.
8. Execute below lines to populate db with test data
```python
import factory
import factory.fuzzy
from taxonomy import models
from test_utils.factories import *
XBlockSkillsFactory.create_batch(5)
SkillFactory.create_batch(20)
XBlockSkillDataFactory.create_batch(20, skill=factory.Iterator(models.Skill.objects.all()), xblock=factory.Iterator(models.XBlockSkills.objects.all()), verified=factory.fuzzy.FuzzyChoice([True, False]))
CourseSkillsFactory.create_batch(5, skill=factory.Iterator(models.Skill.objects.all()))
```
9. Goto http://localhost:18381/api-docs/ in browser.
10. Play with `/taxonomy/api/v1/skills/` & `/taxonomy/api/v1/xblocks/` API.

**Merge checklist:**
    - `make upgrade && make requirements` have been run to regenerate requirements
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [ ] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.
